File: codewars/6kyu/mexican-wave/dduck-mexican-wave.js
// mexican-wave
// https://www.codewars.com/kata/58f5c63f1e26ecda7e000029

const wave = (str) => str.split('')
  .map((v, i, arr) => {
    if (v !== ' ') {
      return [...arr.slice(0, i), v.toUpperCase(), ...arr.slice(i + 1)].join('');
    }
  }).filter((v) => v);

result = ["Hello", "hEllo", "heLlo", "helLo", "hellO"];
console.log(wave("hello"), result, "Should return: '" + result + "'");

result = ["Codewars", "cOdewars", "coDewars", "codEwars", "codeWars", "codewArs", "codewaRs", "codewarS"];
console.log(wave("codewars"), result, "Should return: '" + result + "'");

result = [];
console.log(wave(""), result, "Should return: '" + result + "'");

result = ["Two words", "tWo words", "twO words", "two Words", "two wOrds", "two woRds", "two worDs", "two wordS"];
console.log(wave("two words"), result, "Should return: '" + result + "'");

result = [" Gap ", " gAp ", " gaP "];
console.log(wave(" gap "), result, "Should return: '" + result + "'");

File: codewars/6kyu/mexican-wave/dduck-mexican-wave.py
# mexican-wave
# https://www.codewars.com/kata/58f5c63f1e26ecda7e000029

from unittest import TestCase


def wave(people):
    result = []

    for i in range(0, len(people)):
        if people[i] != ' ':
            result.append(people[0:i] + people[i].upper() + people[i + 1::])

    return result


result = ["Hello", "hEllo", "heLlo", "helLo", "hellO"]
TestCase().assertEqual(wave("hello"), result)

result = ["Codewars", "cOdewars", "coDewars", "codEwars", "codeWars", "codewArs", "codewaRs", "codewarS"]
TestCase().assertEqual(wave("codewars"), result)

result = []
TestCase().assertEqual(wave(""), result)

result = ["Two words", "tWo words", "twO words", "two Words", "two wOrds", "two woRds", "two worDs", "two wordS"]
TestCase().assertEqual(wave("two words"), result)

result = [" Gap ", " gAp ", " gaP "]
TestCase().assertEqual(wave(" gap "), result)
